docs/cms/input-types.adoc

```diff
@@ -91,22 +91,45 @@ image::Content-selector-default.png[An input box with a dropdown options below i
   </config>
 </input>
 ----
-<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType Samples>>
+<1> *allowContentType* is used to limit the content types that may be selected using the <<allowContentType>>
 <2> *allowPath* is used to limit the path of the content that may be selected using the <<allowPath Samples>>. By default, Content Selector
 only displays content from the same site.
 <3> *treeMode* toggles between tree and flat modes. Default value is `false` (flat mode). +
 image:Content-selector-tree.png[Input box with a dropdown with a tree structure, 350]
 <4> *hideToggleIcon* hides the toggle icon when set to `false`. +
 image:Content-selector-no-icon.png[Input box without an icon, 500]
 
-=== allowContentType Samples
+[#allowContentType]
+=== Allow content type pattern
+There are two modes of matching content type patterns in XP: `LEGACY` to simulate pattern matching before image:../images/xp-770.svg[XP 7.7.0,opts=inline] and `MATCH` as current default mode. It can be configured in <<../deployment/config.adoc#admin, config>>.
 
+==== Legacy +
+
+If the pattern value can be found in any part of the content type name, then it is considered a match.
+The pattern will be prefixed by the current application name unless:
+- the pattern uses the <application-name:content-name> format
+- the pattern includes one of `:` or `*`
+- the pattern starts with `${app}`.
+
+==== Match +
+A pattern will be treated as Java Regular Expression (with the exception to Special handling) and the whole content type name must match it to be accepted.
+The pattern will be prefixed by the current application name if it contains none of the following characters `\^$.|?*+()[{:`.
+
+==== Special handling
+
+.`${app}` template expression
+Most content type names start with the name of the application where they are stored. It may be quite messy and hard to maintain the whole application name in every pattern, so the '${app}' template expression can be used to replace it.
+
+.'*' character
```

Comment:
Just further to the conversation we had previously regarding quotes: If possible, I think we should be consistent in the uses of quotes. These lines and the next one mix single quotes and backticks. As before, the choice between single quotes or backticks is up to you, but I suggest something like the following:

```suggestion
.`${app}` template expression
Most content type names start with the name of the application where they are stored. It may be quite messy and hard to maintain the whole application name in every pattern, so the `${app}` template expression can be used to replace it.

.`*` character
```